Source/Core/destroyObject.js

```diff
@@ -23,7 +23,7 @@ define([
      *
      * @exports destroyObject
      *
-     * @param {Object} object The object to destroy.
+     * @param {*} object The object to destroy.
```

Comment:
The wording implies this should still be an `Object`, I don't think you can destroy primitive types.